communication/mavlink_waypoint_handler.hpp

```diff
@@ -256,20 +150,46 @@ class Mavlink_waypoint_handler
     void control_time_out_waypoint_msg();
 
     /**
-     * \brief   Sends the travel time between the last two waypoints
+     * \brief   Gets the current waypoint index
      *
-     * \param   mavlink_stream          The pointer to the MAVLink stream structure
-     * \param   msg                     The pointer to the MAVLink message
+     * \return  Current waypoint index
      */
-    void send_nav_time(const Mavlink_stream* mavlink_stream, mavlink_message_t* msg);
-
+    uint16_t current_waypoint_index() const;
 
     /**
-     * \brief   Initialise the position hold mode in Dubin navigation
+     * \brief   Sets the current waypoint index if possible
      *
-     * \param   local_pos               The position where the position will be held
+     * \param   index   The new waypoint index
+     *
+     * \return  Success
      */
-    void dubin_hold_init(local_position_t local_pos);
+    bool set_current_waypoint_index(int index);
+
+protected:
+    Waypoint waypoint_list_[MAX_WAYPOINTS];                     ///< The array of all waypoints (max MAX_WAYPOINTS)
+    uint16_t waypoint_count_;                                   ///< The total number of waypoints
+    uint16_t current_waypoint_index_;                           ///< The current waypoint index
+
+    const Mavlink_stream& mavlink_stream_;                      ///< The reference to MAVLink stream object
+    INS& ins_;                                                  ///< The pointer to the position estimation structure
+    State& state_;                                              ///< The reference to the state object
+    Navigation& navigation_;                                    ///< The reference to the navigation object
+    Mavlink_message_handler& message_handler_;                  ///< The reference to the mavlink message handler
+    Mission_handler_registry& mission_handler_registry_;        ///< The reference to the mission handler registry
```

Comment:
done in https://github.com/lis-epfl/MAVRIC_Library/pull/331/commits/4ef5f075ac569e4cb6a29ae887491475c582d2c0, https://github.com/lis-epfl/MAVRIC_Library/pull/331/commits/bdfe2486808ebb41fcf3a44c93fc7efc141cd223 and https://github.com/lis-epfl/MAVRIC_Library/pull/331/commits/b1791cb1678991ed4d202c3641c652b205b45a9b